### ISSUE TITLE
Is it possible to implement these features from ai toolkit into kohya. Fully explained down.

### ISSUE THREAD

**Issue by DarkViewAI** [kohya-ss/sd-scripts#1538]
I have done over 300+ trainings now (renting gpus) of different training parameters from both kohya and ai toolkit. 

I get good to decent results now on both.

I believe if you implement these kohya can improve. Im not sure if im right or wrong, but please let me know.

1. noise_scheduler: flowmatch  ||||||||| with the sampler - sampler: flowmatch
not sure what this does
2. content_or_style: balanced
seems to give a good balance when prompting
3. quantize: true
i tested quantize and the quality is basically close to the same, but with even lower vram usage to use.

4. multires training
hopefully an easier way to implement it where it just takes your one dataset and buckets it into 512,768,1024 or whichever res it needs to bucket correctly.

 5. linear_timesteps: true
 not sure what this does but apprently it improves the model

**Comment by peteallen** [comment]
For multi-res training, might it be better if the behavior was that it would only include each image once, in the largest bucket it can be in without upscaling? That way you wouldn't be effectively adding repeats or training on downscaled images when a higher-res version is available.

**Comment by wogam** [comment]
> > 2. content_or_style: balanced
> > seems to give a good balance when prompting
> 
> This is same as `--timestep_sampling sigmoid`.
> 
> > 4. multires training
> > hopefully an easier way to implement it where it just takes your one dataset and buckets it into 512,768,1024 or whichever res it needs to bucket correctly.
> 
> This is possible even now. Although it is a little complicated, it allows for more flexible settings such as changing the batch size. See https://github.com/kohya-ss/sd-scripts/tree/sd3?tab=readme-ov-file#flux1-multi-resolution-training
> 
> > 5. linear_timesteps: true
> > not sure what this does but apprently it improves the model
> 
> I haven't checked the details, but it's probably the same as `--timestep_sampling uniform`.
> 
> Other points are unclear until I check the implementation. It will take some time.

With the dataset config files, is it possible to set a different LR for each resolution? E.g. 512 BS 8 LR learning_rate = 8e-4, 768 BS 4 learning_rate = 4e-4

**Comment by WarAnakin** [comment]
> > > 2. content_or_style: balanced
> > >    seems to give a good balance when prompting
> > 
> > 
> > This is same as `--timestep_sampling sigmoid`.
> > > 4. multires training
> > >    hopefully an easier way to implement it where it just takes your one dataset and buckets it into 512,768,1024 or whichever res it needs to bucket correctly.
> > 
> > 
> > This is possible even now. Although it is a little complicated, it allows for more flexible settings such as changing the batch size. See https://github.com/kohya-ss/sd-scripts/tree/sd3?tab=readme-ov-file#flux1-multi-resolution-training
> > > 5. linear_timesteps: true
> > >    not sure what this does but apprently it improves the model
> > 
> > 
> > I haven't checked the details, but it's probably the same as `--timestep_sampling uniform`.
> > Other points are unclear until I check the implementation. It will take some time.
> 
> With the dataset config files, is it possible to set a different LR for each resolution? E.g. 512 BS 8 LR learning_rate = 8e-4, 768 BS 4 learning_rate = 4e-4

no. In order to do that, you need to start a new training with a different learn rate.

**Comment by luyuhua** [comment]
> > 2. content_or_style: balanced
> > seems to give a good balance when prompting
> 
> This is same as `--timestep_sampling sigmoid`.
> 
> > 4. multires training
> > hopefully an easier way to implement it where it just takes your one dataset and buckets it into 512,768,1024 or whichever res it needs to bucket correctly.
> 
> This is possible even now. Although it is a little complicated, it allows for more flexible settings such as changing the batch size. See https://github.com/kohya-ss/sd-scripts/tree/sd3?tab=readme-ov-file#flux1-multi-resolution-training
> 
> > 5. linear_timesteps: true
> > not sure what this does but apprently it improves the model
> 
> I haven't checked the details, but it's probably the same as `--timestep_sampling uniform`.
> 
> Other points are unclear until I check the implementation. It will take some time.

> content_or_style: balanced
> seems to give a good balance when prompting
> This is same as --timestep_sampling sigmoid.
hi,you mean "balanced" is same as --timestep_sampling sigmoid?
if content_or_style: style,how to set --timestep_sampling?
I understand that content_or_style will use different loss functions for different layers, but I'm not sure if my understanding is correct.
